lib/src/detectors/virtualbox_detector.cc

```diff
@@ -1,21 +1,18 @@
 #include <internal/detectors/virtualbox_detector.hpp>
-#include <leatherman/logging/logging.hpp>
-#include <leatherman/execution/execution.hpp>
 #include <leatherman/util/regex.hpp>
 
 using namespace std;
 using namespace whereami;
-using namespace leatherman::execution;
 using namespace leatherman::util;
 
 namespace whereami { namespace detectors {
 
-    bool virtualbox(const sources::dmi_base& dmi_source) {
+    bool virtualbox(const sources::cpuid_base& cpuid_source,
+                    const sources::dmi_base& dmi_source) {
         static const boost::regex re_virtualbox{"[Vv]irtual[Bb]ox"};
 
-        return re_search(
-            dmi_source.product_name(),
-            re_virtualbox);
+        return cpuid_source.vendor() == "VBoxVBoxVBox" ||
+            re_search(dmi_source.product_name(), re_virtualbox);
```

Comment:
longer-term we're going to want to be able to track which data sources were successful, but this is good until we have a metadata object to work with :+1: